src/math/MathUtils.js

```diff
@@ -229,6 +229,69 @@ function setQuaternionFromProperEuler( q, a, b, c, order ) {
 
 }
 
+function denormalize( value, array ) {
+
+	switch ( array.constructor ) {
+
+		case Float32Array:
+
+			return value;
+
+		case Uint16Array:
+
+			return value / 65535.0;
+
+		case Uint8Array:
+
+			return value / 255.0;
+
+		case Int16Array:
+
+			return Math.max( value / 32767.0, -1.0 );
+
+		case Int8Array:
+
+			return Math.max( value / 127.0, -1.0 );
+
+		default:
+
+			throw new Error( 'Invalid component type.' );
+
+	}
+
+}
+
+function normalize( value, array ) {
+
+	switch ( array.constructor ) {
```

Comment:
Would it be better to add a `scale` member to the class instead and do this switch in the constructor? Then we could just multiply/divide by the `scale` on every access instead of needing the branches?

Reply:
@donmccurdy @Mugen87 Does this sound reasonable? Should I make an alternate PR for it?

Reply:
Yes, sounds like a good alternative!

Reply:
Note that if we want to give the same results as hardware, they're not just scale/division factors, there's a section (search for "int-to-float") in the glTF spec providing the exact expressions:

https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#:~:text=use%20following%20equations%20to%20decode%20real

Reply:
Thanks, that's good to know. Still, I think for our cases (hit tests, bounding boxes) we're unlikely to have identical output to the GPU anyway since floating point error will also accumulate in other places as the code paths are not identical.

Reply:
The correct version is not necessarily longer, we could set two internal function members in the BufferAttribute  constructor:

```js
this._normalizeScalar = someFunc; // (value: number) => number
this._denormalizeScalar = someFunc; // (value: number) => number
```

Reply:
- Alternative PR: #24087